### DISCUSSION TITLE
How to differentiate between 2 finger drag and 2 finger pinch

### DISCUSSION THREAD

**Discussion by Rajdeepc** [pmndrs/use-gesture#505] [Q&A]
In onDrag and onPinch both dragging and pinching are true and its difficult to manage both the functionalities together. Need some help here.

**Comment by dbismut** [comment]
Two fingers drag is not supported. You can probably use the movement of the pinch origin to handle the drag movement but that's not supported out of the box.

**Reply by dbismut** [reply]
I mean computing yourself the displacement of `state.origin`. That implies using memo, pseudo code below:

```js
usePinch( ({ origin: [ox, oy], memo = [ox, oy] }) => {
  const dragMovement = [ox - memo[0], oy - memo[1]]
  return memo 
})
```

**Reply by Rajdeepc** [reply]
@dbismut  Well this is helpful. 
But as of now I am able to pinch on an object with canvas and also drag the canvas around with 2 fingers individually. These two functionalities are working but the below code works for drag only and if I comment out drag pinch works. 
So now pinch and drag doesnt work together. Whenever I try to pinch it drags the canvas. This is my issue at the moment.

**Reply by dbismut** [reply]
This goes beyond the scope of what I believe QA for this lib is intended for as I think it goes to implementation details from your very specific logic. I unfortunately don't have time to go deep into individual projects.
